app/components/pdfViewer/index.tsx

```diff
@@ -286,34 +293,72 @@ const PDFViewer: React.FunctionComponent<PDFViewerProps> = props => {
                     <RelatedPapers shouldShowRelatedPapers={shouldShowRelatedPapers} />
                   </>
                 ) : (
-                  <ScinapseButton
-                    gaCategory="PDF viewer"
-                    gaAction={actionTag}
-                    style={readAllBtnStyle}
-                    content={
-                      <span>
-                        READ ALL <Icon icon="ARROW_POINT_TO_UP" className={styles.arrowIcon} />
-                      </span>
+                  <ClickAwayListener
+                    onClickAway={() =>
+                      dispatch(
+                        ActionCreators.togglePDFBlockedPopper({
+                          isOpenBlockedPopper: false,
+                        })
+                      )
                     }
-                    isLoading={PDFViewerState.isLoading}
-                    disabled={PDFViewerState.hasFailed}
-                    onClick={async () => {
-                      const isBlocked = await blockUnverifiedUser({
-                        authLevel: AUTH_LEVEL.VERIFIED,
-                        actionArea: "pdfViewer",
-                        actionLabel: actionTag,
-                        userActionType: actionTag,
-                      });
-
-                      trackClickButton(actionTag, props.paper.id);
-
-                      if (isBlocked) {
-                        return;
-                      }
+                  >
+                    <div ref={viewMorePDFBtnEl}>
+                      <ScinapseButton
+                        gaCategory="PDF viewer"
+                        gaAction={actionTag}
+                        style={readAllBtnStyle}
+                        content={
+                          <span>
+                            READ ALL <Icon icon="ARROW_POINT_TO_UP" className={styles.arrowIcon} />
+                          </span>
+                        }
+                        isLoading={PDFViewerState.isLoading}
+                        disabled={PDFViewerState.hasFailed}
+                        onClick={async () => {
```

Comment:
pull out this logic to other function.
this pattern makes the content be difficult to read.